test/vaadin-radio-group.html

```diff
@@ -381,6 +381,30 @@
         }, 100);
       });
 
+      it('should set focused attribute on focusin event dispatched', () => {
+        vaadinRadioButtonList[0].dispatchEvent(new CustomEvent('focusin', {composed: true, bubbles: true}));
+        expect(vaadinRadioButtonGroup.hasAttribute('focused')).to.be.true;
+      });
+
+      it('should set focused attribute on radio-button focus', () => {
+        vaadinRadioButtonList[0].focus();
+        expect(vaadinRadioButtonGroup.hasAttribute('focused')).to.be.true;
+      });
+
+      it('should not set focused attribute on focusin event dispatched when disabled', () => {
+        vaadinRadioButtonGroup.disabled = true;
+        vaadinRadioButtonList[0].dispatchEvent(new CustomEvent('focusin', {composed: true, bubbles: true}));
+        expect(vaadinRadioButtonGroup.hasAttribute('focused')).to.be.false;
+      });
+
+      if (!/Trident/.test(navigator.userAgent)) { // Skip IE11, fails in sauce labs
+        it('should remove focused attribute on radio-button blur', () => {
+          vaadinRadioButtonList[0].dispatchEvent(new CustomEvent('focusin', {composed: true, bubbles: true}));
+          vaadinRadioButtonList[0].blur();
```

Comment:
Ahem, this uses a synthetic `'focusin'` event, and then a native `.blur()`.

This is fragile: the synthetic event is not guaranteed to make a native focus on the element. If that indeed didn’t happen, then a native blur could have trouble removing a non-existent native focus too.

I think we should be consistent about this, so that in scope of a single test, we use either both native `.focus()` and `.blur()` methods (although there could be other issues with those), or use both synthetic `'focusin'` and `'focusout'` events.

Would suggest synthetic events, they are generally more reliable in testing, that way we could avoid issues caused by lack of focus on the browser window when running the tests, for example.

Reply:
@platosha fixed.